ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java

```diff
@@ -115,7 +115,7 @@ static void processCompactionLogs(String dir, String storeId, StoreKeyFactory st
     File[] files =
         storeDir.listFiles((fileDir, name) -> name.startsWith(storeId + COMPACTION_LOG_SUFFIX + BlobStore.SEPARATOR));
     List<File> sortedFiles = Stream.of(files)
-        .sorted((file1, file2) -> (int) (getStartTimeFromFile(file2) - getStartTimeFromFile(file1)))
+        .sorted((file1, file2) -> getStartTimeFromFile(file2) - getStartTimeFromFile(file1) > 0 ? 1 : -1)
```

Comment:
Will we hit transitive problem, if two files has the same time?

Reply:
Two files wouldn't have the same time since there will only be one compaction at a time. If this happened, then we need to have some one to take a look.